Northstar.CustomServers/mod/scripts/vscripts/mp/_base_gametype_mp.gnut

```diff
@@ -235,6 +243,40 @@ void function CodeCallback_OnPlayerRespawned( entity player )
 	ClearLastAttacker( player ) // so dying to anything doesn't credit the same attacker after respawning
 }
 
+void function GracePeriodThink( entity player )
+{
+	thread GracePeriodThink_Threaded( player )
+}
+
+void function GracePeriodThink_Threaded( entity player )
+{
+	player.EndSignal( "OnDestroy" )
+	player.EndSignal( "OnPrimaryAttack" )
+	player.EndSignal( "OnDeath" )
+
+	OnThreadEnd(
+		function(): ( player )
+		{
+			if( IsValid( player ) && IsAlive( player ) && GetGameState() == eGameState.Playing ) // if player died we don't reset this( mostly because round end )
+			{
+				player.s.inGracePeriod = false
+				player.Signal( "GracePeriodDone" )
+			}
+		}
+	)
+
+	float countingTime
```

Comment:
```suggestion
	float countingTime = Time() + CLASS_CHANGE_GRACE_PERIOD
```

Reply:
I was using `countingTime` is because I want to make players inside of dropship won't have their loadout selection time reduced( they will have full 20s to swap loadout after they jump out of the dropship ), but it doesn't seem necessary so I'm gonna remove it